process/transaction/baseProcess.go

```diff
@@ -218,16 +218,40 @@ func (txProc *baseTxProcessor) VerifyTransaction(tx *transaction.Transaction) er
 	return txProc.checkTxValues(tx, senderAccount, receiverAccount, false)
 }
 
+// Setting a guardian is allowed with regular transactions on a guarded account
+// but in this case is set with the default epochs delay
+func checkOperationAllowedToBypassGuardian(txData []byte) error {
+	if process.IsSetGuardianCall(txData) {
+		return nil
+	}
+
+	return process.ErrOperationNotPermitted
```

Comment:
by returning this error here, do we consider this transaction as invalid (included in an invalid block) and consumed all the gas? Or it is completely removed from the pool as it was never broadcast?

Reply:
same also for line 267?
if the scammer signs transactions with a different guardian he could pay the fees. 

`if !bytes.Equal(guardian, tx.GuardianAddr) {
		return process.ErrTransactionAndAccountGuardianMismatch
	}`

Reply:
wondering if this can be an attack vector as the scammer increments the account's nonce

Reply:
It should not be possible to spend the gas, as this is a bad transaction and does not consume fee.
Did some change to remove these transactions also from the txpool.
If a guarded account has a pending guardian already set, then transactions to replace the pending guardian will be ignored.

Reply:
moved the verification to (txProc *baseTxProcessor) checkTxValues so that it covers also scheduled txs